### PR TITLE
[HIPIFY] Packaging for hipify_clang

### PR DESCRIPTION
**Pull request by aakanksha555**

### CODE REVIEW

File: packaging/hipify-clang.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(hipify-clang)
+
+install(PROGRAMS @PROJECT_BINARY_DIR@/hipify-clang/hipify-clang DESTINATION bin)
+
+#############################
+# Packaging steps
+#############################
+set(CPACK_SET_DESTDIR TRUE)
+set(CPACK_INSTALL_PREFIX "/opt/rocm/hip")
```
Comment (aakanksha555) [inline packaging/hipify-clang.txt]: What would be the CPACK_INSTALL_PREFIX for hipify-clang?

File: CMakeLists.txt [hunk 2]
```diff
@@ -433,6 +433,18 @@ if (BUILD_HIPIFY_CLANG)
...
+# Package: hipify-clang
+set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/packages/hipify-clang)
+configure_file(packaging/hipify-clang.txt ${BUILD_DIR}/CMakeLists.txt @ONLY)
+add_custom_target(pkg_hipify-clang  COMMAND ${CMAKE_COMMAND} .
+    COMMAND rm -rf *.deb *.rpm *.tar.gz
+    COMMAND make package
+    COMMAND cp *.deb ${PROJECT_BINARY_DIR}
+    COMMAND cp *.rpm ${PROJECT_BINARY_DIR}
+    COMMAND cp *.tar.gz ${PROJECT_BINARY_DIR}
+    WORKING_DIRECTORY ${BUILD_DIR}
```
Comment (emankov) [inline CMakeLists.txt]: Missing closing bracket.

Comment (emankov) [inline CMakeLists.txt]: I believe it should go into hipify-clang's CMakeLists.txt.

File: packaging/hipify-clang.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(hipify-clang)
+
+install(PROGRAMS @PROJECT_BINARY_DIR@/hipify-clang/hipify-clang DESTINATION bin)
```
Comment (emankov) [inline packaging/hipify-clang.txt]: If a common practice is to duplicate installation steps, those steps should be taken from https://github.com/ROCm-Developer-Tools/HIP/blob/master/hipify-clang/CMakeLists.txt#L83-L115. 

What about clang header files, being installed with hipify-clang?

Btw, could you test packaging without this particular line?

File: packaging/hipify-clang.txt (hunk 1 above)
Comment (emankov) [reply packaging/hipify-clang.txt]: 1. `hipify-clang` is a multi-platform tool, but as long as `HIP` itself and its cmake building/installing system are not multi-platform yet, we may not, I guess, add Windows support for hipify-clang at a packaging step for now.
2. As for `CPACK_INSTALL_PREFIX` for `hipify-clang` for `UNIX` platforms, well, it is a question. As far as hipify-clang is going to be shipped with `hip-clang`, not with `HIP`, what is CPACK_INSTALL_PREFIX for `hip-clang`?

Comment (emankov) [reply packaging/hipify-clang.txt]: > What would be the CPACK_INSTALL_PREFIX for hipify-clang?
1. `hipify-clang` is a multi-platform tool, but as long as `HIP` itself and its cmake building/installing system are not multi-platform yet, we may not, I guess, add `Windows` support for `hipify-clang` at a packaging step for now.
2. As for `CPACK_INSTALL_PREFIX` for `hipify-clang` for `UNIX` platforms, well, it is a question. As far as `hipify-clang` is going to be shipped with `hip-clang`, not with `HIP`, what is `CPACK_INSTALL_PREFIX` for `hip-clang`?

**Comment by emankov** [comment]
@aakanksha555, just a gentle ping, any updates?

File: CMakeLists.txt (hunk 2 above)
Comment (aakanksha555) [reply CMakeLists.txt]: Moved it to hipify-clang's CMakeLists.txt.

File: packaging/hipify-clang.txt (hunk 1 above)
Comment (aakanksha555) [reply packaging/hipify-clang.txt]: 1. Sure..
2. Where can I find the CPACK_INSTALL_PREFIX for hip-clang?

File: hipify-clang/CMakeLists.txt [hunk 4]
```diff
@@ -114,6 +114,34 @@ install(
...
+for i in *.rpm ; do mv \$i \${i/.rpm/.x86_64.rpm} ; done
+")
+add_custom_target(package
+    COMMAND bash ${PROJECT_BINARY_DIR}/fixnames
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    DEPENDS pkg_hipify-clang)
+if(POLICY CMP0037)
+    cmake_policy(POP)
+endif()
+
```
Comment (emankov) [inline hipify-clang/CMakeLists.txt]: It contains `UNIX` related staff. hipify-clang also supports Windows. As long as we are not shipping it on Windows yet, better to add `if (UNIX) ... endif()` wrapper for the whole packaging staff.

File: hipify-clang/packaging/hipify-clang.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(hipify-clang)
 
-install(PROGRAMS @PROJECT_BINARY_DIR@/hipify-clang/hipify-clang DESTINATION bin)
+install(PROGRAMS @PROJECT_BINARY_DIR@/hipify-clang DESTINATION bin)
```
Comment (emankov) [inline hipify-clang/packaging/hipify-clang.txt]: I do not understand the necessity of install procedure call here, is install commands in hipify-clang's CMakeLists.txt not enough?

File: CMakeLists.txt [hunk 6]
```diff
@@ -433,6 +433,7 @@ if (BUILD_HIPIFY_CLANG)
     add_dependencies(pkg_hip_base hipify-clang)
 endif()
 
+
```
Comment (emankov) [inline CMakeLists.txt]: No need in touching this file.

File: hipify-clang/CMakeLists.txt (hunk 4 above)
Comment (aakanksha555) [reply hipify-clang/CMakeLists.txt]: Okay, sounds good.

**Comment by emankov** [comment]
@aakanksha555, just a gentle ping. What's up with packaging hipify-clang activity?

**Comment by aakanksha555** [comment]
@emankov can you continue the conversation on the other latest patch: https://github.com/aakanksha555/HIP/pull/2

**Pull request closed without merging**